mlflow/models/evaluation/default_evaluator.py

```diff
@@ -569,6 +569,12 @@ def plot_feature_importance():
             "shap_feature_importance_plot",
         )
 
+    def _evaluate_sklearn_model_score_if_needed(self):
+        if self.model_loader_module == "mlflow.sklearn":
+            from sklearn.model_selection._search import BaseSearchCV
+            if isinstance(self.raw_model, BaseSearchCV) and self.raw_model.scoring is not None:
+                self.metrics["score"] = self.raw_model.score(self.X, self.y)
```

Comment:
Does `self.raw_model.score` always score a scalar?

Reply:
Looks like It does:
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV.score

Reply:
It should. Only real numbers can be compared unambiguously. See make scorer score_fn signature: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.make_scorer.html

Reply:
```python
from sklearn import svm, datasets
from sklearn.model_selection import GridSearchCV

iris = datasets.load_iris()
parameters = {"kernel": ("linear", "rbf"), "C": [1, 10]}
svc = svm.SVC()
clf = GridSearchCV(svc, parameters)

m = clf.fit(iris.data, iris.target)
print(m.scoring)
print(m.score(iris.data, iris.target))
```

```
None
0.9933333333333333
```

`score` also works when `self.scoring` is None.

Reply:
```
from sklearn import svm, datasets
from sklearn.model_selection import GridSearchCV

iris = datasets.load_iris()
parameters = {"kernel": ("linear", "rbf"), "C": [1, 10]}
svc = svm.SVC()
clf = GridSearchCV(svc, parameters, refit=False)

m = clf.fit(iris.data, iris.target)
print(m.scoring)
print(m.score(iris.data, iris.target))
```

```
...
AttributeError: This GridSearchCV instance was initialized with `refit=False`. score is available only after refitting on the best parameters. You can refit an estimator manually using the `best_params_` attribute
```

`score` throws when `refix` is set to `False`. Can we avoid calling `score` if the model is not refitted?

Reply:
> score also works when self.scoring is None. Do we always want to log it ?

Good question. I think we can always log it.

Reply:
Note all sklearn estimator support "score" method. shall we log score for all estimators ? 
and we can swallow exception raised by `model.score` and skip logging it when it failed.

Reply:
This is a good idea. We could just try to execute the estimator's score method, log the result if the method is available and score is computed, or log a warning otherwise.

Reply:
+1 to Jin's suggestion